src/main/anovos/data_analyzer/association_eval_varclus.py

```diff
@@ -0,0 +1,388 @@
+        Parameters
+        ----------
+        df
+            Spark dataframe
+        clus1
+            List of feature columns in first cluster
+        clus2
+            List of feature columns in second cluster
+        feat_list
+            List of features to re-assign
+
+        Returns
+        -------
+        fin_clus1, fin_clus2, max_var
+            fin_clus1: final cluster 1's list of feature columns
+            fin_clus2: final cluster 2's list of feature columns
+            max_var: the maximum variance achieved
+
+        """
+
+        if feat_list is None:
+            feat_list = clus1 + clus2
+
+        init_var = self._calc_tot_var(df, clus1, clus2)[0]
+        fin_clus1, fin_clus2 = clus1[:], clus2[:]
+        check_var, max_var = (init_var,) * 2
+        while True:
+
+            for feat in feat_list:
+                new_clus1, new_clus2 = fin_clus1[:], fin_clus2[:]
+                if feat in new_clus1:
+                    new_clus1.remove(feat)
+                    new_clus2.append(feat)
+                elif feat in new_clus2:
+                    new_clus1.append(feat)
+                    new_clus2.remove(feat)
+                else:
+                    continue
+                new_var = self._calc_tot_var(df, new_clus1, new_clus2)[0]
+                if new_var > check_var:
+                    check_var = new_var
+                    fin_clus1, fin_clus2 = new_clus1[:], new_clus2[:]
+
+            if max_var == check_var:
+                break
+            else:
+                max_var = check_var
+        return fin_clus1, fin_clus2, max_var
+
+    def _reassign_rs(self, df, clus1, clus2, n_rs=0):
+        """
+        This function should be called after _reassign, and it performs random shuffling of n_rs times to run the re-assignment
+        Parameters
+        ----------
+        df
+            Spark dataframe
+        clus1
+            List of feature columns in first cluster
+        clus2
+            List of feature columns in second cluster
+        n_rs
+            Number of random shuffles
+
+        Returns
+        -------
+        fin_rs_clus1, fin_rs_clus2, max_rs_var
+            fin_rs_clus1: final cluster 1's list of feature columns after random shuffling
+            fin_rs_clus2: final cluster 2's list of feature columns after random shuffling
+            max_rs_var: the maximum variance achieved after random shuffling
+        """
+
+        feat_list = clus1 + clus2
+        fin_rs_clus1, fin_rs_clus2, max_rs_var = self._reassign(df, clus1, clus2)
+
+        for _ in range(n_rs):
+            random.shuffle(feat_list)
+            rs_clus1, rs_clus2, rs_var = self._reassign(df, clus1, clus2, feat_list)
+            if rs_var > max_rs_var:
+                max_rs_var = rs_var
+                fin_rs_clus1, fin_rs_clus2 = rs_clus1, rs_clus2
+
+        return fin_rs_clus1, fin_rs_clus2, max_rs_var
+
+    def _varclusspu(self):
```

Comment:
As above, we can include `spark` argument in here, instead of in `init`. 

One more thing, _varclusspu is named by the VarClusHi author, stands for "speed-up" version. We can rename this here, to something that makes more sense and distinguish our function, something like `_varclusspark`.